server/resp/src/main/java/org/infinispan/server/resp/RespHandler.java

```diff
@@ -86,23 +89,36 @@ private void flushAccessLog(ChannelHandlerContext ctx, ChannelPromise promise, C
       accessLogger.flush(ctx, promise, res);
    }
 
+   protected RespRequestHandler initializeRespRequestHandler() {
+      AdvancedCache<byte[], byte[]> cache = respServer.getCache();
+      if (respServer.getConfiguration().authentication().enabled()) {
+         return new Resp3AuthHandler(respServer, cache);
+      }
+
+      return respServer.newHandler(cache);
+   }
+
    @Override
    public void channelRegistered(ChannelHandlerContext ctx) throws Exception {
       ctx.channel().attr(RespRequestHandler.BYTE_BUF_POOL_ATTRIBUTE_KEY)
             .set(size -> allocateBuffer(ctx, size));
+      requestHandler = initializeRespRequestHandler();
       this.accessLogger = traceAccess
             ? new AccessLoggerManager(ctx, requestHandler.respServer().getTimeService())
             : null;
       requestHandler.respServer().metadataRepository().client().incrementConnectedClients();
+      resumeAutoRead(ctx);
```

Comment:
Must admit these calls are not sitting right. We shouldn't have a read call to resume while processing the registration of a channel.

Reply:
This is the order of channel creation https://livebook.manning.com/book/netty-in-action/chapter-6/ch06fig01. Note that you should not be receiving any data until after the channel is completely active.

Reply:
I think you were going to add a comment here as to why this is needed? My guess is this is because this is not actually a registered call and the handlers are added after the channel has been started like I mention in my other comment.